### DISCUSSION TITLE
Change sphere color

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#4423] [Q&A] [answered]
Using this piece of code i wanted to add a sphere to the scene with a white color
`builder.setSymbology(ColorDef.white, ColorDef.white, 1);
    builder.addSolidPrimitive(Sphere.createCenterRadius(point, 1))`

But as you can see the color of the sphere is always this grey color
![image](https://user-images.githubusercontent.com/29980658/193853082-ba3024c2-fb6a-4e4e-beca-2b299e332cc1.png)

**Comment by pmconne** [comment]
Provide full code please.

**Reply by salah-walid** [reply]
```typescript
export default class Graphic implements Decorator {
  private readonly point : Point3d

  public constructor(point: Point3d) {
   this.point = point
  }

  public readonly useCachedDecorations = true;

  public decorate(context: DecorateContext): void {
    if (context.viewport.view.isSpatialView()) {
      const branch= new GraphicBranch(true)
      const graphics= context.createGraphic({
        type: GraphicType.WorldDecoration,
        generateEdges: true,
      })

      //Creating the sphere
      builder.setSymbology(ColorDef.white, ColorDef.white, 1);
      builder.addSolidPrimitive(Sphere.createCenterRadius(this.point, 0.6))

      branch.setViewFlagOverrides({visibleEdges: true});
      branch.add(graphic.finish())
      context.addDecoration(GraphicType.WorldDecoration, context.createBranch(branch, Transform.identity));
    }
  }
}

 ```

**Comment by bbastings** [comment, marked as answer]
Looks like a white sphere affected by the view lighting...

**Reply by salah-walid** [reply]
Is there a way to give it a plain color ?

**Reply by pmconne** [reply]
You can tell us not to [produce normal vectors](https://www.itwinjs.org/reference/core-frontend/rendering/graphicbuilderoptions/wantnormals/) for it. That will prevent it from being lit.
However, an unlit sphere is nearly indistinguishable from a circle, but more expensive to draw.

**Reply by salah-walid** [reply]
Can the specular radius be reduced at least ?

**Reply by pmconne** [reply]
Yes, replace your use of `setSymbology` with a call to [activateGraphicParams](https://www.itwinjs.org/reference/core-frontend/rendering/graphicbuilder/activategraphicparams/) supplying a GraphicParams with a custom [RenderMaterial](https://www.itwinjs.org/reference/core-common/rendering/rendermaterial/).

**Reply by pmconne** [reply]
If you want to reduce the specular for the entire view, change the [light settings](https://www.itwinjs.org/reference/core-common/displaystyles/displaystyle3dsettings/lights/).
